presto-elasticsearch/src/main/java/com/facebook/presto/elasticsearch/ElasticsearchClient.java

```diff
@@ -0,0 +1,374 @@
+    {
+        if (tableDescription.getIndexExactMatch()) {
+            return ImmutableList.of(tableDescription.getIndex());
+        }
+        TransportClient client = clients.get(tableDescription.getClusterName());
+        verify(client != null, "TransportClient is null");
+        String[] indices = client.admin()
+                .indices()
+                .getIndex(new GetIndexRequest())
+                .actionGet(timeout.toMillis())
+                .getIndices();
+        return Arrays.stream(indices)
+                    .filter(index -> index.startsWith(tableDescription.getIndex()))
+                    .collect(toImmutableList());
+    }
+
+    public ClusterSearchShardsResponse getSearchShards(String index, ElasticsearchTableDescription tableDescription)
+    {
+        TransportClient client = clients.get(tableDescription.getClusterName());
+        verify(client != null, "TransportClient is null");
+        ClusterSearchShardsRequest request = new ClusterSearchShardsRequest(index);
+        return client.admin()
+                .cluster()
+                .searchShards(request)
+                .actionGet(timeout.toMillis());
+    }
+
+    private List<ColumnMetadata> buildMetadata(List<ElasticsearchColumn> columns)
+    {
+        List<ColumnMetadata> result = new ArrayList<>();
+        for (ElasticsearchColumn column : columns) {
+            Map<String, Object> properties = new HashMap<>();
+            properties.put("jsonPath", column.getJsonPath());
+            properties.put("jsonType", column.getJsonType());
+            properties.put("isList", column.isList());
+            properties.put("ordinalPosition", column.getOrdinalPosition());
+            result.add(new ColumnMetadata(column.getName(), column.getType(), "", "", false, properties));
+        }
+        return result;
+    }
+
+    private List<ElasticsearchColumn> buildColumns(ElasticsearchTableDescription tableDescription)
+    {
+        List<ElasticsearchColumn> columns = new ArrayList<>();
+        TransportClient client = clients.get(tableDescription.getClusterName());
+        verify(client != null, "TransportClient is null");
+        for (String index : getIndices(tableDescription)) {
+            try {
+                GetMappingsRequest mappingsRequest = new GetMappingsRequest().types(tableDescription.getType());
+
+                if (!isNullOrEmpty(index)) {
+                    mappingsRequest.indices(index);
+                }
+                ImmutableOpenMap<String, ImmutableOpenMap<String, MappingMetaData>> mappings = client.admin()
+                        .indices()
+                        .getMappings(mappingsRequest)
+                        .actionGet(timeout.toMillis())
+                        .getMappings();
+
+                Iterator<String> indexIterator = mappings.keysIt();
+                while (indexIterator.hasNext()) {
+                    // TODO use io.airlift.json.JsonCodec
```

Comment:
could you please address this?

Reply:
Please create an issue for this (with some details). Even if you cannot find time to address that, someone in the community can help.

Reply:
sure, will do